### DISCUSSION TITLE
preserveOnce

### DISCUSSION THREAD

**Discussion by roydekleijn** [cypress-io/cypress#23639] [Questions and Help] [answered]
Hello,

I norices `preserveOnce` is deprecated and there is some reference to the session documentation. But how should this look like if we implement it according `session`?

~~~javascript
  cy.getCookies().then(cookies => {
    const namesOfCookies = cookies.map(c => c.name)
    Cypress.Cookies.preserveOnce(...namesOfCookies)
  })
~~~

Any help would be much appreciated. Maybe we can eventually update the documentation,
Roy

**Comment by emilyrohrbough** [comment, marked as answer]
@roydekleijn `cy.session()` will preserve the cookies for you so using the command similar to the following will persist these cookies for you:
```
beforeEach(() => {
  cy.session('unique_id', () => {
    // logic to establish the cookies you need to preserver
  })
})

it('', () => {
  // cookies have been applied from saved session in before-each hook
  cy.visit('url_to_test')
  // ....
})
```

**Reply by roydekleijn** [reply]
I think my case is a bit different. We have a very large SPA application. I want to implement some logical "teststeps" in each `it`. So within a single `describe` all the `it`s should continue on the same session, without doing cy.visit() everytime.  (is that possible?)

The deprecated way of preserving cookies works perfectly for my case. But ya, it's deprecated...

**Reply by jozso39** [reply]
I recommend turning off `testIsolation` to tests, you have to specify it on the top describe of the test:
```
describe(`your test`, { testIsolation: false }, () => {
		it(`this test or step`, () => {
			// test code
		});
	});
	
```

This way, the session will not be cleared between it-s.
There is another problem: the sessions will carry over to another tests. You just have to clear the session in the before hook.
